numpy/tests/test_public_api.py

```diff
@@ -109,3 +117,341 @@ def test_NPY_NO_EXPORT():
+    "core.shape_base",
+    "core.umath",
+    "core.umath_tests",
+    "distutils.ccompiler",
+    "distutils.command",
+    "distutils.command.autodist",
+    "distutils.command.bdist_rpm",
+    "distutils.command.build",
+    "distutils.command.build_clib",
+    "distutils.command.build_ext",
+    "distutils.command.build_py",
+    "distutils.command.build_scripts",
+    "distutils.command.build_src",
+    "distutils.command.config",
+    "distutils.command.config_compiler",
+    "distutils.command.develop",
+    "distutils.command.egg_info",
+    "distutils.command.install",
+    "distutils.command.install_clib",
+    "distutils.command.install_data",
+    "distutils.command.install_headers",
+    "distutils.command.sdist",
+    "distutils.compat",
+    "distutils.conv_template",
+    "distutils.core",
+    "distutils.extension",
+    "distutils.fcompiler",
+    "distutils.fcompiler.absoft",
+    "distutils.fcompiler.compaq",
+    "distutils.fcompiler.environment",
+    "distutils.fcompiler.g95",
+    "distutils.fcompiler.gnu",
+    "distutils.fcompiler.hpux",
+    "distutils.fcompiler.ibm",
+    "distutils.fcompiler.intel",
+    "distutils.fcompiler.lahey",
+    "distutils.fcompiler.mips",
+    "distutils.fcompiler.nag",
+    "distutils.fcompiler.none",
+    "distutils.fcompiler.pathf95",
+    "distutils.fcompiler.pg",
+    "distutils.fcompiler.sun",
+    "distutils.fcompiler.vast",
+    "distutils.from_template",
+    "distutils.info",
+    "distutils.intelccompiler",
+    "distutils.lib2def",
+    "distutils.line_endings",
+    "distutils.mingw32ccompiler",
+    "distutils.msvc9compiler",
+    "distutils.msvccompiler",
+    "distutils.npy_pkg_config",
+    "distutils.numpy_distribution",
+    "distutils.pathccompiler",
+    "distutils.unixccompiler",
+    "f2py.auxfuncs",
+    "f2py.capi_maps",
+    "f2py.cb_rules",
+    "f2py.cfuncs",
+    "f2py.common_rules",
+    "f2py.crackfortran",
+    "f2py.diagnose",
+    "f2py.f2py2e",
+    "f2py.f2py_testing",
+    "f2py.f90mod_rules",
+    "f2py.func2subr",
+    "f2py.info",
+    "f2py.rules",
+    "f2py.use_rules",
+    "fft.helper",
+    "fft.info",
+    "fft.pocketfft",
+    "fft.pocketfft_internal",
+    "lib.arraypad",
+    "lib.arraysetops",
+    "lib.arrayterator",
+    "lib.financial",
+    "lib.function_base",
+    "lib.histograms",
+    "lib.index_tricks",
+    "lib.info",
+    "lib.nanfunctions",
+    "lib.npyio",
+    "lib.polynomial",
+    "lib.shape_base",
+    "lib.stride_tricks",
+    "lib.twodim_base",
+    "lib.type_check",
+    "lib.ufunclike",
+    "lib.user_array",  # note: not in np.lib, but probably should just be deleted
+    "lib.utils",
+    "linalg.info",
+    "linalg.lapack_lite",
+    "linalg.linalg",
+    "ma.bench",
+    "ma.core",
+    "ma.testutils",
+    "ma.timer_comparison",
+    "ma.version",
+    "matrixlib",
+    "matrixlib.defmatrix",
+    "random.bit_generator",
+    "random.bounded_integers",
+    "random.common",
+    "random.entropy",
```

Comment:
This conflicts with gh-14498, whichever is merged first then the other will have to remove this line.

Reply:
Also: `random.info` does not exist. Do the tests check that all these actually are present?

Reply:
No, they were generated from an install and the checks are for no additions, not for removals (that'd be either a good thing or caused by tests, so less interesting to test from my perspective).

Reply:
> Also: `random.info` does not exist.

this is not true, there's a `numpy/random/info.py` in current master

Reply:
removed `random.entropy`

Reply:
I think it would be preferable if the tests check whether they are exactly present (unless there is some reason that is unreliable). It is good to know when we remove it, and when we do, we want to make sure we do not accidentally readd.

Reply:
Okay done. That did identify one more stray file to clean up, and will help keep things clean.

Possible this isn't 100% robust, since importing may fail on different platforms with different exceptions - let's deal with that if/when such cases show up.

Reply:
CI all green on four platforms without anything unexpected, so robustness probably okay. comments here addressed, so resolving this one.